src/board.cpp

```diff
@@ -0,0 +1,230 @@
+{  
+    block = Block(newType);
+    Pos = startingPos;
+}
+
+int Board::getRowPos()
+{
+    return Pos.row;
+}
+
+int Board::getColPos()
+{
+    return Pos.col;
+}
+
+Block Board::getBlock()
+{
+    return block;
+}
+
+bool Board::getCell(Position pos)
+{
+    return board[pos.row][pos.col];
+}
+
+BlockType Board::getBlockType(Position pos)
+{
+    return blockType[pos.row][pos.col];
+}
+
+void Board::clear(Position pos)
+{
+    board[pos.row][pos.col] = false;
+    blockType[pos.row][pos.col] = (BlockType) 0;
+}
+
+void Board::clear(int row)
+{
+    for(uint col = 0; col < WIDTH; ++col)   clear(Position(row, col));
+}
+
+void Board::clear()
+{
+    for(uint row = 0; row < HEIGHT; ++row)  clear(row);
+}
+
+bool Board::isCellFree(Position pos)
+{
+    return !board[pos.row][pos.col];
+}
+
+bool Board::isRowFree(int row)
+{
+    for(uint col = 0; col < WIDTH; ++col)
+        if(!isCellFree(Position(row, col)))   return false;
+
+    return true;
+}
+
+bool Board::isRowFull(int row)
+{
+    for(uint col = 0; col < WIDTH; ++col)
+        if(isCellFree(Position(row, col)))   return false;
+
+    return true;
+}
+
+bool Board::isLegal(Block newBlock, Position newPos)   // sprawdzanie czy klocek o zadanej pozycji nie koliduje z planszą
+{
+    std::vector<std::vector<bool>> blockMatrix = newBlock.getMatrix();
+
+    for(uint row = 0; row < blockMatrix.size(); ++row)
+    {
+        for(uint col = 0; col < blockMatrix[row].size(); ++col)
+        {
+            if(!blockMatrix[row][col])  continue;
+
+            Position posOnBoard(newPos.row + row, newPos.col + col);
+
+            if(posOnBoard.row < 0 || posOnBoard.row >= HEIGHT) return false;
+
+            if(posOnBoard.col < 0 || posOnBoard.col >= WIDTH) return false;
+
+            if(!isCellFree(posOnBoard))   return false;
+        }   
+    }
+
+    return true;
+}
+
+bool Board::canAddBlock(BlockType type) // czy można dodać klocek na pozycję startową, jeśli nie można no to wiadomo koniec gry
+{
+    return isLegal(Block(type), startingPos);
+}
+
+bool Board::addBlock(BlockType type)    // dodawanie klocka na pozycję startową (false - nie można dodać)
+{
+    if(canAddBlock(type))
+    {
+        setNewBlock(type);
+        return true;
+    }
+    return false;
+}
+
+bool Board::canMoveBlock(MoveDirection direction)
+{
+    Position newPos = Pos;
+    newPos.move(direction);
+
+    return isLegal(block, newPos);
+}
+
+bool Board::moveBlock(MoveDirection direction)  
+{
+    if(canMoveBlock(direction))
+    {   // ruszenie klocka na zadaną pozycję
+        Pos.move(direction);
+        return true;
+    }
+    return false;
+}
+
+bool Board::canRotateBlock(RotationDirection direction)
+{
+    Block blockCopy = block;    // kopiowanie klocka, żeby można było zrobić spokojnie obrót, sprawdzić czy jeste poprawy a dopiero potem go zastosować
+    blockCopy.rotate(direction);
+
+    return isLegal(blockCopy, Pos);
+}
+
+bool Board::rotateBlock(RotationDirection direction)
```

Comment:
Nazwij to jakoś attemptToRotateBlock, bo ta funkcja niekoniecznie obróci klocek, a tylko spróbuje